test_case/ERA5TimeSeriesDataset.py

```diff
@@ -6,13 +6,47 @@
 2. PyTorchERA5Dataset: A wrapper class to use the custom dataset in PyTorch DataLoader.
 """
 import os
-import xarray as xr
+
 import torch
-from torch.utils.data import Dataset, DataLoader
+import xarray as xr
+from nvidia.dali import fn
+from nvidia.dali.pipeline import pipeline_def
+from torch.utils.data import Dataset
+
+
+def _load_data(
+    data_path: str = "/glade/derecho/scratch/ksha/CREDIT_data/ERA5_mlevel_arXiv",
+    start_year: int = 2013,
+    end_year: int = 2014,
+    input_vars: list[str] = ["t2m", "V500", "U500", "T500", "Z500", "Q500"],
+) -> xr.Dataset:
+    """Loads all zarr files into a dictionary keyed by year."""
+    zarr_paths: list[str] = []
+    for year in range(start_year, end_year + 1):
+        zarr_path: str = os.path.join(
+            data_path, f"SixHourly_y_TOTAL_{year}-01-01_{year}-12-31_staged.zarr"
+        )
+        if os.path.exists(zarr_path):
+            zarr_paths.append(zarr_path)
+        else:
+            print(f"Data for year {year} not found!!!")
+    ds: xr.Dataset = xr.open_mfdataset(
+        zarr_paths, engine="zarr", consolidated=True, combine="by_coords"
+    )[input_vars]
+    # self.length = ds.sizes["time"]
+    return ds
+
 
+@pipeline_def
+def zarr_pipeline():
+    array = fn.python_function(function=_load_data, device="cpu")
```

Comment:
Might need to use [`functools.partial`](https://docs.python.org/3/library/functools.html#functools.partial) as a hack to be able to pass in custom arguments (e.g. data_path, start_year, end_year, etc) to `_load_data`?